Mathlib/FieldTheory/RatFunc/AsPolynomial.lean

```diff
@@ -207,6 +210,61 @@ theorem valuation_of_mk (f : Polynomial K) {g : Polynomial K} (hg : g ≠ 0) :
       (Polynomial.idealX K).intValuation f / (Polynomial.idealX K).intValuation g := by
   simp only [RatFunc.mk_eq_mk' _ hg, valuation_of_mk']
 
+end HeightOneSpectrum
+
+section TrivialOnConstants
+
+variable {Γ : Type*} [LinearOrderedCommMonoidWithZero Γ] {v : Valuation (RatFunc K) Γ}
+
+open Valuation
+
+/--
+If a valuation is trivial on constants then the valuation of `(monomial n a)` is equal
+to `(v (RatFunc.X)) ^ n`. -/
+lemma valuation_monomial_eq_valuation_X_pow
+    (hv : ∀ a : K, a ≠ 0 → v (algebraMap K (RatFunc K) a) = 1) (n : ℕ) {a : K} (ha : a ≠ 0) :
+    v (monomial n a) = v RatFunc.X ^ n := by
+  have h : algebraMap _ (RatFunc K) (C a) = algebraMap _ _ a := rfl
+  simp only [RatFunc.coePolynomial, ← C_mul_X_pow_eq_monomial, _root_.map_mul, _root_.map_pow,
+    RatFunc.algebraMap_X, h, hv a ha, one_mul]
+
+/--
+If a valuation is trivial on constants and `1 < v RatFunc.X` then
+`v p = v RatFunc.X ^ p.natDegree`. -/
+theorem valuation_eq_valuation_X_pow_natDegree_of_one_lt_valuation_X
+    [Nontrivial Γ] [PosMulStrictMono Γ] (hlt : 1 < v RatFunc.X)
+    (h : ∀ a : K, a ≠ 0 → v (algebraMap K (RatFunc K) a) = 1) {p : Polynomial K} (hp : p ≠ 0) :
+    v p = v RatFunc.X ^ p.natDegree := by
+  rw [RatFunc.coePolynomial,
+    ← valuation_monomial_eq_valuation_X_pow _ h _ (leadingCoeff_ne_zero.mpr hp)]
+  nth_rw 1 [as_sum_range p, map_sum]
+  apply map_sum_eq_of_lt _ (by simp) (by simp [hp])
+  intro i hi
+  simp only [Finset.mem_sdiff, Finset.mem_range, Finset.mem_singleton, Nat.lt_add_one_iff,
+    ← lt_iff_le_and_ne] at hi
+  simp_all only [ne_eq, RatFunc.algebraMap_eq_C, ← C_mul_X_pow_eq_monomial, _root_.map_mul,
+    RatFunc.algebraMap_C, _root_.map_pow, RatFunc.algebraMap_X, coeff_natDegree,
+    leadingCoeff_eq_zero, not_false_eq_true, one_mul, gt_iff_lt]
+  by_cases h0 : (p.coeff i) = 0
+  · simp only [h0, _root_.map_zero, zero_mul, one_mul]
+    exact pow_pos (lt_trans zero_lt_one hlt) p.natDegree
+  simp only [h _ h0, one_mul, pow_lt_pow_right₀ hlt hi]
+
+/--
+If a valuation is trivial on constants and `v RatFunc.X ≤ 1` then `v p ≤ 1`. -/
```

Comment:
Same comment as above both about the inequality and the docs (add "polynomial").